framework/aster-frame/src/sync/mutex.rs

```diff
@@ -77,6 +77,12 @@ impl<'a, T> MutexGuard<'a, T> {
     }
 }
 
+impl<'a, T> MutexGuard<'a, T> {
+    pub fn get_lock(&self) -> &'a Mutex<T> {
```

Comment:
Smart pointer types `P` like `Box`, `Arc`, `Mutex` implements `Deref<T>`. To avoid method name conflict between `P` and `T`, the author of type `P` usually does not add methods to `P`. For example, `Arc` defines `strong_count` as an associated function.

```rust
pub fn strong_count(this: &Arc<T, A>) -> usize
```

Rather than a method.

```rust
pub fn strong_count(&self) -> usize
```

We should do the same for `get_lock`.